### ISSUE TITLE
Improve readability of "assert failed" message for DictionaryContainsKeyValuePairConstraint WithValue()

### ISSUE THREAD

**Issue by GustaMagik** [nunit/nunit#3718]
Request for improvement of the assert message for .WithValue(), introduced in #3486 

**Current**
```
Message: 
      Expected: dictionary containing entry System.Collections.DictionaryEntry
      But was:  < ["firstname", "Jane"], ["lastname", "Doe"] >
```
      
**Preferred**
```
Message: 
      Expected: dictionary containing entry ["firstname", "John"]
      But was:  < ["firstname", "Jane"], ["lastname", "Doe"] >
```
      
This would bring the message more in line with that of Contains.Key / Contains.Value
```
Message:
      Expected: dictionary containing key "email"
      But was:  < ["firstname", "Jane"], ["lastname", "Doe"] >
```

**Comment by mikkelbu** [comment]
I think this improvement makes sense.

**Comment by manfred-brands** [comment]
I stumbled into the same issue today and was about to create a ticket. Luckily I looked first.

`System.Collection.DictionaryEntry` has no `ToString()` overload and hence prints the type name.

Am I right that all that needs doing is add a formatter to `MsgUtils` like:

```
AddFormatter(next => val => val is DictionaryEntry de ? FormatKeyValuePair(de.Key, de.Value) : next(val));
```

It seems to work:

> Error Message:
     Expected: dictionary containing entry [123, <Classes.NUnit.Class1Fixture>]
  But was:  < [123, <Classes.Class1>], [234, <Classes.Class2>] >

I will raise a PR.
